src/DynamoCoreWpf/Utilities/MessageBoxUtilities.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System.Windows;
+
+namespace Dynamo.Wpf.Utilities
+{
+    // Wrapper over MessageBox.Show
+    // Useful for testing
+    internal class MessageBoxService {
+        internal interface IMessageBox
+        {
+            MessageBoxResult Show(string msg, string title, MessageBoxButton button, MessageBoxImage img);
+        }
+
+        // Default implementation of the IDialogService interface
+        // Use this instead of MessageBox.Show
+        internal class DefaultMessageBox : IMessageBox
+        {
+            MessageBoxResult IMessageBox.Show(string msg, string title, MessageBoxButton button, MessageBoxImage img)
+            {
+                return MessageBox.Show(msg, title, button, img);
+            }
+        }
+
+        private static IMessageBox msg_box;
+
+        // Use this method to override the internal IMessageBox interface with a mocked implementation.
+        internal static void OverrideMessageBoxDuringTests(IMessageBox msgBox) { msg_box = msgBox; }
```

Comment:
Why can't we use the `moq` library for this in the tests, why do we need to implement our own classes to mock the message box?

Reply:
`moq` only works on interfaces. The only way to moq the MessageBox class is to wrap it in an interface.